cumulus/client/cirrus-executor/src/processor.rs

```diff
@@ -202,8 +203,26 @@ where
 			header_hash
 		);
 
-		let execution_receipt =
-			ExecutionReceipt { primary_hash, secondary_hash: header_hash, trace, trace_root };
+		let primary_number = self
```

Comment:
We already have block number available in the place that eventually leads to here, I think would be nicer to add `primary_number` as an argument here (or maybe the whole `ActivatedLeaf`, not sure).

Reply:
Agree, I'll probably go with adding another argument `primary_number` as the name `ActivatedLeaf` sounds a little bit foreign outside overseer and I don't have a better naming :P.

Reply:
Updated in https://github.com/subspace/subspace/pull/427/commits/0ef1d9495f112c2b76e46d2dea4256e27e9cb777